### ISSUE TITLE
Allowing decompose the SOAP descriptor by atomic species pairs 

### ISSUE THREAD

**Issue by wushanyun64** [SINGROUP/dscribe#38]
Hi,

Since SOAP vector is defined as a vector containing elements p(r){Z1,Z2,n,n',l}. I'm wondering is there an easy way to decompose the vector into sub-vectors containing certain (Z1, Z2) combination? Based on the SOAP vector we have now, it seems that the vector is sequenced as: {Z1(Z1, Z2, Z3.....), Z2(Z2, Z3, Z4....), Z3....}, am I understanding this right? If so, how can we make sure which section of the sector corresponds to which pair of elements?  Thanks in advance!

**Comment by lauri-codes** [comment]
Hi @wushanyun64,

You can see how the output of SOAP is ordered [at the tutorial page.](https://singroup.github.io/dscribe/tutorials/soap.html) (see the pseudo-code block on that page). Essentially you're right in that the output is ordered into pairs of species.

Would you think that implementing a function that would return the indices of specific species-combinations would serve your purpose? We already provide this kind of function for MBTR ([see here](https://singroup.github.io/dscribe/tutorials/mbtr.html#locating-information)), so to stay consistent we could also implement if for SOAP and ACSF. What do you think?

**Comment by wushanyun64** [comment]
> Hi @wushanyun64,
> 
> You can see how the output of SOAP is ordered [at the tutorial page.](https://singroup.github.io/dscribe/tutorials/soap.html) (see the pseudo-code block on that page). Essentially you're right in that the output is ordered into pairs of species.
> 
> Would you think that implementing a function that would return the indices of specific species-combinations would serve your purpose? We already provide this kind of function for MBTR ([see here](https://singroup.github.io/dscribe/tutorials/mbtr.html#locating-information)), so to stay consistent we could also implement if for SOAP and ACSF. What do you think?

Hi, Lauri：
If we can have a function like `get_location()`  in SOAP, that would be very helpful! I imagine something like :

```
soap.get_location((Zi,Zj)):
    return ((a,b) which indicates the slice corresponds to the location of P{Zi, Zj})
```
Also, one question, since based on the SOAP tutorial,  in the SOAP vector both P{Zi, Zj} and P{Zj, Zi} will be calculated, how you will address this in the function `get_location()`?  Thank you!

**Comment by lauri-codes** [comment]
I'm now preparing a bigger update for version 0.3.0 and I can include `get_location()` in it. I will publish it within a few days.

> Also, one question, since based on the SOAP tutorial, in the SOAP vector both P{Zi, Zj} and P{Zj, Zi} will be calculated, how you will address this in the function get_location()? Thank you!

Actually only one of the identical pairs (I think the one where` i<=j`) is included in the SOAP output. The `get_location()`-function would accept both (Z_i, Z_j) and (Z_j, Z_i) as arguments, but would return the same location. This is also how it works for MBTR.

**Comment by wushanyun64** [comment]
That's great, can't waiting to try the new version.

**Comment by lauri-codes** [comment]
Hi @wushanyun64,

The `get_location()`-method can now be found in the development version 0.3.0a0 that can be installed from the git master branch. (just to `git clone git@github.com:SINGROUP/dscribe.git` and `pip install .` within the cloned folder)

I'm still workin on some features for the 0.3.0 release. When that is ready, you can find this same feature on version 0.3.0.) Message me here if you find any issues with the implementation.

**Comment by lauri-codes** [comment]
The `get_location` function is now available in version 0.3.1 which is available through `pip`.
